changes.go

```diff
@@ -162,27 +162,44 @@ func (g *Commands) resolveChangeListRecv(
 			return
 		}
 	}
-
-	// TODO: limit the number of active tasks for children lookups
 	dirlist := merge(remoteChildren, localChildren)
+
+	// Arbitrary value. TODO: Calibrate or calculate this value
+	chunkSize := 100
+	srcLen := len(dirlist)
+	chunkCount, remainder := srcLen/chunkSize, srcLen%chunkSize
+	i := 0
+
+	if remainder != 0 {
+		chunkCount += 1
+	}
+
 	var wg sync.WaitGroup
-	wg.Add(len(dirlist))
-	for _, l := range dirlist {
-		go func(wg *sync.WaitGroup, isPush bool, cl *[]*Change, p string, l *dirList) {
+	wg.Add(chunkCount)
```

Comment:
So this means it's switching from len(dirlist) go-routines operating in parallel to len(dirlist)/100? From my messing around earlier, I think that will mean no parallelism in most cases (for my account at least, I was mostly seeing < 100 but with occasional big ones). It would be good to include a comment outlining the intended parallelism and why.

Another strategy would be to make a channel and have a fixed number of go-routines pull from the channel. Then just push all the dirlist entries down that channel

http://jmoiron.net/blog/limiting-concurrency-in-go/

so then there's always a fixed, small number of workers and you always get parallelism.